lte/cloud/go/services/subscriberdb/storage/sub_proto.go

```diff
@@ -0,0 +1,335 @@
+	errs := &multierror.Error{}
+	for _, subProto := range subProtos {
+		marshaled, err := proto.Marshal(subProto)
+		if err != nil {
+			multierror.Append(errs, errors.Wrapf(err, "serialize subproto of ID %+v", lte_protos.SidString(subProto.Sid)))
+			continue
+		}
+		insertQuery = insertQuery.Values(network, lte_protos.SidString(subProto.Sid), marshaled)
+	}
+	if errs.ErrorOrNil() != nil {
+		return errs
+	}
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		_, err := insertQuery.RunWith(tx).Exec()
+		return nil, errors.Wrapf(err, "insert sub protos into store for network %+v", network)
+	}
+	_, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	return err
+}
+
+// CommitUpdateByNetwork cleans up and re-populates the subscriber proto store table with data from the temporary
+// table for a particular network, and then truncates the temporary table.
+func (l *SubProtoStore) CommitUpdateByNetwork(network string) error {
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		// HACK: hard coding sql query because there currently doesn't exist good support
+		// for this following query with squirrel
+		deleteRowsOnlyInRealTableQuery := fmt.Sprintf(
+			"DELETE FROM %s WHERE (%s = \"%s\") AND ((%s, %s) NOT IN (SELECT %s, %s FROM %s))",
+			subProtoTableName, subProtoNidCol, network,
+			subProtoNidCol, subProtoSidCol,
+			subProtoNidCol, subProtoSidCol, subProtoTmpTableName,
+		)
+		_, err := tx.Exec(deleteRowsOnlyInRealTableQuery)
+		if err != nil {
+			return nil, errors.Wrapf(err, "clean up previous sub proto store table")
+		}
+
+		// the upsertColumnPrefix is used to refer to the the table whose column value is
+		// used in the "on conflict update" operation; it depends on the sql dialect
+		// e.g. for mariaDB, "f1=t2.f1"; for postgres, "f1=excluded.f1"
+		var upsertColumnPrefix string
+		switch getSqlDialect() {
+		case sqorc.PostgresDialect:
+			upsertColumnPrefix = postgresUpsertColumnPrefix
+		case sqorc.MariaDialect:
+			upsertColumnPrefix = subProtoTmpTableName
+		}
+		_, err = l.builder.
+			Insert(subProtoTableName).
+			Select(
+				l.builder.
+					Select(subProtoNidCol, subProtoSidCol, subProtoProtoCol).
+					From(subProtoTmpTableName).
+					Where(squirrel.Eq{subProtoNidCol: network}),
+			).
+			OnConflict(
+				[]sqorc.UpsertValue{{
+					Column: subProtoProtoCol,
+					Value:  squirrel.Expr(fmt.Sprintf("%s.%s", upsertColumnPrefix, subProtoProtoCol)),
+				}},
+				subProtoNidCol, subProtoSidCol,
+			).
+			RunWith(tx).
+			Exec()
+		if err != nil {
+			return nil, errors.Wrapf(err, "populate sub proto store table")
+		}
+
+		_, err = l.builder.
+			Delete(subProtoTmpTableName).
+			Where(squirrel.Eq{subProtoNidCol: network}).
+			RunWith(tx).
+			Exec()
+		if err != nil {
+			return nil, errors.Wrapf(err, "clean up tmp sub proto store table")
+		}
+		return nil, nil
+	}
+	_, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	return err
+}
+
+func (l *SubProtoStore) ClearTmpTable() error {
```

Comment:
"InitiateUpdate" -- the fact that we're clearing a tmp table should be an implementation detail. Instead, it can be "InitiateUpdate", do some work, then "ApplyUpdate"

And put the fns in order -- "InitateUpdate" should be higher in the file than "ApplyUpdate"